### PR TITLE
HGCal - bugfix - removing energy of out-of-range hits

### PR DESCRIPTION
**Pull request by felicepantaleo**
This PR is a bug fix and should solve the problem in #20458 
Hits which are outside the 10cm radius from the center of gravity are removed by setting their fraction to zero, but their energy was never subtracted from the energy of the cluster.
This is a problem as the final energy of the PFCluster is not computed as the sum of the energy of the hits * their fraction, but it gets assigned the energy of the realistic cluster.
Recalibration is also included
@rovere @kpedro88

### CODE REVIEW

**Comment by felicepantaleo** [comment]
type bug fix

**Comment by felicepantaleo** [comment]
type bugfix

**Comment by kpedro88** [comment]
assign upgrade

**Comment by felicepantaleo** [comment]
It's good to go now.

**Comment by kpedro88** [comment]
please test

**Comment by slava77** [comment]
In the PR description:
> This PR requires recalibration of the PFClusters which should come later tonight

it looks like the calibration was added already.
If nothing else is missing, please update the PR description.

**Comment by felicepantaleo** [comment]
@slava77 done

**Comment by kpedro88** [comment]
+1
minor fluctuations in HGCal supercluster quantities, as expected for 10 events
also, tested w/ #20458 w/ `dropUnseedable` set to false, and the problematic event runs successfully

**Comment by slava77** [comment]
@felicepantaleo  @predragm
since this required recalibration, please attach/post some plots in the PR description that confirm the response with this PR is good.

File: RecoParticleFlow/PFClusterProducer/python/particleFlowRealisticSimClusterHGCCalibrations_cfi.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 
 minEtaCorrection = cms.double(1.4)
 maxEtaCorrection = cms.double(3.0)
-hadronCorrections = cms.vdouble(1.16, 1.10, 1.08, 1.08, 1.09, 1.11, 1.14, 1.26)  
-egammaCorrections = cms.vdouble(1.00, 1.00, 1.00, 1.00, 1.02, 1.03, 1.03, 1.03)  
+hadronCorrections = cms.vdouble(1.24, 1.24, 1.24, 1.23, 1.24, 1.25, 1.29, 1.29)
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/python/particleFlowRealisticSimClusterHGCCalibrations_cfi.py]: is it clear how the supposedly minor effect of the fix is affecting the hadron correction so significantly.
Some eta bins are up by 15% (compared to at most 1% for egamma corrs).

Comment (felicepantaleo) [reply RecoParticleFlow/PFClusterProducer/python/particleFlowRealisticSimClusterHGCCalibrations_cfi.py]: the energy of the hits outside the radius is not accounted anymore, so it's not a minor effect for hadrons.
clusters from 50GeV pions have usually a radius of 10-ish cm (in FH), and are much smaller (2-3cm) in EE.
One could probably set two radii in the future depending on the subdetector...

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/20479	353c6d1
- changes are in line with the description
- jenkins tests pass and comparisons with baseline show changes starting from PF HGCAL supercluster quantities in 2023 workflows: 
    - there are fewer superclusters and the component CaloClusters; 
    - the energy of the component CaloClusters is mostly unchanged in the .energy() variable (driven by hit energies dropped), while the .correctedEnergy() is perturbed more (different calibrations)

Given absence of a backport to 93X, it's unclear if this is required there or can stay in 94X.

**Comment by davidlange6** [comment]
+1

**Pull request merged**